### PR TITLE
TASK: Extract z-index variables into CSS vars

### PR DESCRIPTION
**Pull request by Duske**
This PR is a first draft on how to move the z-indices to CSS vars as discussed in #793.
It reveals some issues which need to be solved:

- [x]  find a proper naming for the z-index-variables, e.g. scope the name to the corresponding component
- [x]  find a proper structure for the style config object
- [x]  refactor the existing [CSS vars](https://github.com/neos/neos-ui/blob/master/packages/build-essentials/src/webpack.config.js#L55) which are currently directly set in the webpack config. Maybe this out of scope of this PR which should deal with z-indices only.

### CODE REVIEW

**Comment by dimaip** [comment]
Great!

> e.g. scope the name to the corresponding component
What do you mean?

> Maybe this out of scope
As it is comfortable for you. I'd just go ahead and do it, it's just a copy paste and would allow to see the structure of the styles object clearer.

**Comment by Duske** [comment]
> e.g. scope the name to the corresponding component

In the configuration object there are the properties `aspectRatioDropdownContents ` and `dropdownContents ` for example. They're basically dealing with the same stuff, but need different names (collision) as they are different components. So we could use some scoping like this:

```js
zIndex:{
  aspectRatio: {
    dropdownContents: '1'
  }
  someOtherComponent: {
    dropdownContents: '2'
  }
}

```

> Maybe this out of scope

Sounds good, just wanted to make sure that you're okay with it :)

**Comment by Duske** [comment]
> e.g. scope the name to the corresponding component

The [configuration object](https://github.com/neos/neos-ui/pull/1124/files#diff-8d8c129d1fbe4d0e5d3cb6cfbcb0074aR4) contains the properties `aspectRatioDropdownContents ` and `dropdownContents `, which are dealing with the same stuff but are in located in different components. They needed a different name because of the flat hierarchy. So we could scope them to the belonging component:
```
zIndex:{
  aspectRatio: {
    dropdownContents: '1'
  }
  someOtherComponent: {
    dropdownContents: '2'
  }
}

```

> Maybe this out of scope

Sounds good, just wanted to make sure you're okay with it :)

**Comment by dimaip** [comment]
oh you know what actually could make sense: group zIndexes around stacking contexts they divide into. Wdyt?

**Comment by dimaip** [comment]
Smth like this could be helpful to understand it: https://chrome.google.com/webstore/detail/z-context/jigamimbjojkdgnlldajknogfgncplbh?hl=en

**Comment by dimaip** [comment]
Also relevant: https://www.smashingmagazine.com/2014/06/sassy-z-index-management-for-complex-layouts/

**Comment by Duske** [comment]
Just to verify that we're on the same page:
We group the various zIndex properties by their stacking contexts. So every element with a z-index which `parentStackingContext ` is `html` is set on the root level of zIndex. The property for these elements is called `context`. Child elements with z-index properties will be put inside.
```
zIndex:{
  dropdownContext: { // new stacking context
    context: '1',
    contents: '2',
    button: '3'
  }
}
```

Do you mean something like that? Should we receive the indices by using lists as described in the post?

**Comment by dimaip** [comment]
Yep. I think it makes total sense. I hate those 99999 indexes really.

**Comment by Duske** [comment]
I'm trying to group them by their stackingContext but its kinda hard. For example sometimes a a checkboxInput component has `html` as parentStackingContext, sometimes it is the surrounding dropdown component. Most of the components seem to have `html` as parentStackingContext, so the z-index is just 1. 
This results in a unnecessary complex structure for these simple cases. You can see the current work of the configuration object in [this gist](https://gist.github.com/Duske/45d4640bf957d1199f677f76b22206ba).

To sum it up: It reveals, just as you assumed, that most of these high indices are not needed. But structuring by their stackingContexts seems to be quite complicated and ambiguous.

**Comment by Duske** [comment]
I've finished the grouping and used arrays to describe the z-index as you pointed out in Slack.
Since I'm not sure where `nodeToolBar`  is used, its z-index was not touched

**Comment by Duske** [comment]
I think thats a good idea :) The names become quite long and a delimiter helps to read it more clearly

File: packages/neos-ui/src/Containers/RightSideBar/Inspector/style.css [hunk 1]
```diff
@@ -19,7 +19,7 @@
     left: 0;
     width: calc(100% - var(--sidebarWidth));
     height: 100%;
-    z-index: 10000;
+    z-index: var(--zIndexUnappliedChangesOverlay);
```
Comment (dimaip) [inline packages/neos-ui/src/Containers/RightSideBar/Inspector/style.css]: This one still doesn't have delimiting dashes

File: packages/neos-ui-inspector/src/SecondaryInspector/style.css [hunk 2]
```diff
@@ -29,5 +29,5 @@
     font-size: 18px;
     border-right-width: 0;
     border-top-width: 0;
-    z-index: 10;
+    z-index: var(--zIndexSecondaryInspectorClose);
```
Comment (dimaip) [inline packages/neos-ui-inspector/src/SecondaryInspector/style.css]: same

**Comment by dimaip** [comment]
@Duske so why is this still WIP?

**Comment by Duske** [comment]
Oops you're right, I've updated the PR accordingly 😶

**Comment by mstruebing** [comment]
@Duske Could you rebase your branch to the current master? I think there is nothing against merging it.

**Comment by markusguenther** [comment]
Resolved the conflicts in a PR. Just because I am not able to modify the PR.

**Comment by dimaip** [comment]
@Duske thanks a lot for your amazing effort! Will try to finally merge it with @markusguenther via his fork: #1361
